### PR TITLE
SASS-9721: Allowances section: Create 'Check your answers' page

### PR DESCRIPTION
**Pull request by PraneethN-Kotaru**

### CODE REVIEW

File: test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+  val taxYear = LocalDate.now.getYear
+  val countryCode = "aus"
+
+  val onwardRoute: Call = Call(
+    "GET",
+    s"/update-and-submit-income-tax-return/property/$taxYear/foreign-property/allowances/$countryCode/complete-yes-no"
+  )
+
+  val zeroEmissionCarAllowanceValue = 300
```
Comment (vivekhmrc) [inline test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala:48]: This should be :

`val zeroEmissionCarAllowance = BigDecimal(300)`

File: test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import viewmodels.govuk.SummaryListFluency
+import views.html.foreign.allowances.ForeignAllowancesCheckYourAnswersView
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class ForeignAllowancesCheckYourAnswersControllerSpec extends SpecBase with SummaryListFluency {
+
+  val taxYear = LocalDate.now.getYear
+  val countryCode = "aus"
```
Comment (vivekhmrc) [inline test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala:41]: This should be `"AUS"`

File: test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      running(application) {
+        val request = FakeRequest(GET, routes.ForeignAllowancesCheckYourAnswersController.onPageLoad(taxYear, countryCode).url)
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.routes.JourneyRecoveryController.onPageLoad().url
+      }
+    }
+//
```
Comment (vivekhmrc) [inline test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala:96]: Why are these tests commented ?

**Review by vivekhmrc — CHANGES_REQUESTED** [review]

File: test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala (hunk 1 above)
Comment (PraneethN-Kotaru) [reply test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala:48]: its already been converted to BigDecimal in the next line

File: test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala (hunk 3 above)
Comment (PraneethN-Kotaru) [reply test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala:96]: they are from ONSUBMIT method, As I have not implemented it, I commented it!

File: test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala (hunk 1 above)
Comment (vivekhmrc) [reply test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala:48]: Why we need 2 lines ?

**Review by vivekhmrc — APPROVED** [review]

**Pull request merged**